test/src/0100-reload-switch-debug/main

```diff
@@ -0,0 +1,128 @@
+cvmfs_test_name="Switch between debug and normal mode during cvmfs reload"
+cvmfs_test_autofs_on_startup=false
+cvmfs_test_suites="quick"
+
+source ./src/0100-reload-switch-debug/setup_teardown
+
+TEST0100_REPO=lhcb.cern.ch
+TEST0100_MOUNTPOINT=/cvmfs/lhcb.cern.ch
+
+CVMFS_TEST_0100_TMPFILE=
+
+mount_with_debuglog() {
+  CVMFS_TEST_DEBUGLOG=$(echo "\"${CVMFS_TEST_0100_TMPFILE}\"")
+
+  echo "   ** Mount_with_debuglog"
+  cvmfs_mount $TEST0100_REPO || return $?
+}
+
+mount_without_debuglog() {
+  CVMFS_TEST_DEBUGLOG=""
+
+  echo "   ** Mount_without_debuglog"
+  cvmfs_mount $TEST0100_REPO || return $?
+}
+
+reload_without_debuglog() {
+  echo "   ** Access some data at $TEST0100_REPO"
+  cat /cvmfs/$TEST0100_REPO/*.sh > /dev/null || return 10
+
+  # assumption: debug log is already there
+  echo "   ** Check non-empty existence of debug log"
+  if [[ -s $CVMFS_TEST_0100_TMPFILE ]]
+  then
+    echo "     *** True"
+  else
+    echo "     *** False"
+    return 11
+  fi
+
+  echo "   ** Remove CVMFS_DEBUGLOG from config"
+  sudo sed -i '/^CVMFS_DEBUGLOG=/d' /etc/cvmfs/default.local
```

Comment:
In the macos test there is a 
```
sed: 1: "/etc/cvmfs/default.local": command c expects \ followed by text
```
error, could be due to https://stackoverflow.com/questions/4247068/sed-command-with-i-option-failing-on-mac-but-works-on-linux/14813278#14813278

Needs doublechecking if this is what causes the test to fail

Reply:
I'd go for
```suggestion
  sudo sed -i'' -e '/^CVMFS_DEBUGLOG=/d' /etc/cvmfs/default.local
```